### PR TITLE
feature/data-passing-with-redux

### PR DESCRIPTION
**Pull request by EgijaLocmele**
Changed data passing from props to using Redux logic

### CODE REVIEW

File: src/features/productDetails/ProductDetails.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
 import { useParams } from 'react-router-dom'
-import { ProductDetailsItem } from './ProductDetailsItem'
+import PropTypes from 'prop-types'
+import ProductDetailsItem from './ProductDetailsItem'
 
-export function ProductDetails({ productDetailsData }) {
+const ProductDetails = (props) => {
+  const { products } = props
   let { itemId } = useParams()
```
Comment (YuraLeo94) [inline src/features/productDetails/ProductDetails.js:7]: Why not const? Do you change in this file value of `itemId`?

File: src/features/productList/ProductList.js [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
 import { Link } from 'react-router-dom'
-import { ProductListItem } from './ProductListItem'
+import PropTypes from 'prop-types'
+import ProductListItem from './ProductListItem'
 import './ProductList.scss'
 
-export function ProductList({ productListData }) {
+const ProductList = (props) => {
+  const { products } = props
+  console.log(products)
```
Comment (YuraLeo94) [inline src/features/productList/ProductList.js:9]: Please remove `console.log(products)`

**Review by YuraLeo94 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
 function App() {
+  const product = useSelector((state) => state.productListItems)
+
   return (
     <BrowserRouter>
       <HeaderNavigation />
       <Switch>
         <Route exact path="/">
-          <ProductList productListData={data.productListItems} />
+          <ProductList products={product} />
```
Comment (leleleadele) [inline src/App.js:19]: You're passing only one item to the list or how does this work?

Comment (EgijaLocmele) [reply src/App.js:19]: No, it's basically passing data that is accessed with useSelector(), so not only one item but everything

File: src/features/productDetails/ProductDetails.js [hunk 4]
```diff
@@ -1,9 +1,15 @@
 import { useParams } from 'react-router-dom'
-import { ProductDetailsItem } from './ProductDetailsItem'
+import PropTypes from 'prop-types'
+import ProductDetailsItem from './ProductDetailsItem'
 
-export function ProductDetails({ productDetailsData }) {
-  let { itemId } = useParams()
-  const selectBook = productDetailsData?.find((book) => book.id === itemId)
+const ProductDetails = (props) => {
```
Comment (leleleadele) [inline src/features/productDetails/ProductDetails.js:5]: you can write `const ProductDetails = ({ products }) => ..` instead

File: src/App.js (hunk 3 above)
Comment (leleleadele) [reply src/App.js:19]: The naming implies it's one product though. I suggest to be aware that the naming is understandable to other devs if they look at your app. :)

**Review by leleleadele — APPROVED** [review]

File: src/App.js (hunk 3 above)
Comment (EgijaLocmele) [reply src/App.js:19]: Yeah maybe not the best example for the naming 😄

File: src/features/productDetails/ProductDetails.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
+  const { products } = props
+  const { itemId } = useParams()
+  const selectBook = products?.find((book) => book.id === itemId)
 
-  return <ProductDetailsItem productDetailsData={selectBook} />
+  return <ProductDetailsItem products={selectBook} />
```
Comment (YuraLeo94) [inline src/features/productDetails/ProductDetails.js:8]: Look like you have one `selectBook`, but `products` you have many why?

File: src/features/productDetails/ProductDetailsItem.js [hunk 6]
```diff
@@ -2,22 +2,24 @@ import { Text } from '@fluentui/react/lib/Text'
 import { Image, ImageFit } from '@fluentui/react/lib/Image'
 import { SpinButton, Position } from '@fluentui/react'
 import { PrimaryButton } from '@fluentui/react/lib/Button'
+import PropTypes from 'prop-types'
 import './ProductDetails.scss'
 
-export function ProductDetailsItem({ productDetailsData }) {
+const ProductDetailsItem = (props) => {
+  const { products } = props
```
Comment (YuraLeo94) [inline src/features/productDetails/ProductDetailsItem.js:9]: Why `products`, but not `product`?

File: src/App.js (hunk 3 above)
Comment (YuraLeo94) [reply src/App.js]: I agree with @leleleadele, rename please `product` to `products`

**Review by YuraLeo94 — CHANGES_REQUESTED** [review]

File: src/features/productDetails/ProductDetails.js [hunk 7]
```diff
@@ -1,9 +1,11 @@
...
 
-export function ProductDetails({ productDetailsData }) {
-  let { itemId } = useParams()
-  const selectBook = productDetailsData?.find((book) => book.id === itemId)
+const ProductDetails = ({ products }) => {
+  const { itemId } = useParams()
+  const selectBook = products?.find((book) => book.id === itemId)
 
-  return <ProductDetailsItem productDetailsData={selectBook} />
+  return <ProductDetailsItem products={selectBook} />
```
Comment (YuraLeo94) [inline src/features/productDetails/ProductDetails.js:8]: Look like you have one `selectBook`, but `products` you have many why?

**Review by Jose-Hash-Dev — APPROVED** [review]

**Review by YuraLeo94 — APPROVED** [review]

**Pull request merged**